ehrapy/api/io/read.py

```diff
@@ -110,35 +113,28 @@ def _read(
         return_mudata: bool = False,
         cache: bool = False,
         backup_url: Optional[str] = None,
+        **kwargs,
     ) -> Union[MuData, Dict[str, AnnData], AnnData]:
         """Internal interface of the read method."""
         if cache and return_mudata:
-            raise MudataCachingNotSupportedError(
-                "Caching is currently not supported for MuData objects. Consider setting return_mudata to False in order "
-                "to use caching!"
-            )
+            DataReader._mudata_cache_not_supported()
         # check, whether the datafile(s) is/are present or not
-        is_present = DataReader._check_datafiles_present_and_download(filename, backup_url=backup_url)
-        if not is_present and not filename.is_dir() and not filename.is_file():
-            print(
-                "[bold red]Attempted download of missing dataset file(s) failed. Please file an issue at our repository "
-                "[blue]https://github.com/theislab/ehrapy!"
-            )
-        path_cache_dir = settings.cachedir / filename
+        DataReader._check_files_present(filename, backup_url)
+
+        # multi data format extensions like pdf can contain multiple datasets in one single file and are therefore handled as directories when caching
+        path_cache_dir = settings.cachedir / (
+            filename if filename.suffix not in multi_data_extensions else filename.stem
```

Comment:
Wasn't aware of the `stem`function of Pathlib. Cool!

Reply:
`Pathlib` has some really cool stuff when it comes to work with filenames etc., yeah